### PR TITLE
added a CSRF service provider

### PR DESCRIPTION
**Pull request by fabpot**
Not yet finished.

### CODE REVIEW

**Comment by desarrolla2** [comment]
you can elaborate on what needs to be done?

File: doc/providers/csrf.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    use Silex\Provider\CsrfServiceProvider;
+
+    $app->register(new CsrfServiceProvider());
+
+.. note::
+
+    The *SerializerServiceProvider* relies on Symfony's `Serializer Component
+    <http://symfony.com/doc/current/components/serializer.html>`_, 
+    which comes with the "fat" Silex archive but not with the regular
+    one.  If you are using Composer, add it as a dependency:
```
Comment (stof) [inline doc/providers/csrf.rst]: there are extra spaces here

**Comment by stof** [comment]
The upgrade path must document the need to register the Csrf provider when you were using forms, and the doc for the FormProvider should recommend registering the Csrf provider too. Otherwise your forms are not protected against CSRF.

**Comment by fabpot** [comment]
PR updated, everything should be ok now.

**Pull request merged**